### DISCUSSION TITLE
How to output results on integration points

### DISCUSSION THREAD

**Discussion by pshen20127** [idaholab/moose#24966] [Q&A Modules: Solid mechanics] [answered]
Hello All

Now I output variables in postprocessors in the type of ElementAverageValue. How to output the results on integration points? What type I should choose?
![Weixin Screenshot_20230714092956](https://github.com/idaholab/moose/assets/81499506/cf7b67bc-f6e3-4bb8-86fa-4007e1397bcb)

Thanks
Pengfei

**Comment by GiudGiud** [comment, marked as answer]
Hello

I don’t think we have a quadrature point sampler for variables. However if you use a ParsedMaterial to mirror the variable value (I think stresses are already material properties so you might not even need that) then you can use a 
MaterialVectorPostprocessor (in the VectorPostprocessors block) to sample the variable value indirectly

Guillaume
